octobot_trading/personal_data/portfolios/portfolio_value_holder.py

```diff
@@ -261,21 +291,74 @@ def _try_get_value_of_currency(self, currency, quantity, raise_error):
                     raise missing_data_exception
         return constants.ZERO
 
-    def convert_currency_value_using_last_prices(self, quantity, current_currency, target_currency):
+    def convert_currency_value_using_last_prices(
+        self, quantity, current_currency, target_currency, settlement_asset=None
+    ):
         try:
-            symbol = symbol_util.merge_currencies(current_currency, target_currency)
+            symbol = symbol_util.merge_currencies(
+                current_currency, target_currency, settlement_asset=settlement_asset
+            )
             if self._has_price_data(symbol):
                 return quantity * self._get_last_price_data(symbol)
         except KeyError:
             pass
         try:
-            reversed_symbol = symbol_util.merge_currencies(target_currency, current_currency)
+            reversed_symbol = symbol_util.merge_currencies(
+                target_currency, current_currency, settlement_asset=settlement_asset
+            )
             return quantity / self._get_last_price_data(reversed_symbol)
-        except decimal.DivisionByZero:
-            pass
-        except KeyError:
+        except (KeyError, decimal.DivisionByZero):
             pass
-        raise errors.MissingPriceDataError(f"no price data to evaluate {current_currency} price in {target_currency}")
+        raise errors.MissingPriceDataError(
+            f"no price data to evaluate {current_currency} price in {target_currency}"
+        )
+
+    def try_convert_currency_value_using_multiple_pairs(self, currency, quantity) -> decimal.Decimal:
+        # settlement_asset needs to be handled to add support for futures
+
+        # try with two pairs
+        # for example:
+        # currency: ETH - ref market: USDT
+        # ETH/USDT is not available. ETH/BTC and BTC/USDT are available though.
+        # first convert ETH -> BTC and then BTC -> USDT
+        #               | bridge part 1     | bridge part 2
+
+        # look into available symbols to find pair bridges
+        part_1_base = currency
+        part_2_quote = self.portfolio_manager.reference_market
+        for bridge_part_1_symbol in self.last_prices_by_trading_pair:
```

Comment:
With this code, we will iterate over all trading pairs each time the portfolio value is calculated. I'm not sure it's the best approach, don't you think we could save the "bridge path" to prevent iterating on each traded symbol each time?

Reply:
right, i'll add it